setup/src/Magento/Setup/Module/Di/Code/Reader/FileScanner.php

```diff
@@ -206,8 +207,7 @@ protected function scan()
                 if ($this->tokenType === T_WHITESPACE) {
                     goto SCANNER_NAMESPACE_CONTINUE;
                 }
-
-                if ($this->tokenType === T_NS_SEPARATOR || $this->tokenType === T_STRING) {
+                if (\array_key_exists($this->tokenType, $namespaceContentTokenTypes)) {
```

Comment:
I remember some discussion in Slack about `array_key_exists` and `isset` performance and `isset` looks more faster.
Slack removed chat history but may be @ihor-sviziev has performance test results?

Reply:
Hello @Den4ik 
Thank you for your proposition.
We will fix it.

Reply:
Starting php 7.4 it doesn't make any sense. Since magento 2.4 isn't supporting PHP 7.3 - I think it's ok to use array_key_exists.
https://www.php.net/manual/es/function.array-key-exists.php#125313

PS: to get the maximum performance we should use `\array_key_exists()` to highlight the global namespace.

Reply:
@ihor-sviziev 2.4 still support php 7.3
Do Magento have plans remove 7.3 support?

Reply:
@andrewbess will we drop the php 7.3 support?

Reply:
I believe that 7.3 will be dropped, but not sure that it will be in 2.4 release line

Reply:
@ihor-sviziev PHP 7.3 compatibility will be dropped in 2.4.4